public/app/features/datasources/components/picker/utils.ts

```diff
@@ -39,13 +43,42 @@ export function dataSourceLabel(
   return 'Unknown';
 }
 
-export function useGetDatasources(filters: GetDataSourceListFilters) {
+/**
+ * Stores the uid of the last 5 data sources selected by the user
+ *
+ * @returns function for pushing a data source uid to the store
+ */
+export function useRecentlyUsedDataSources() {
+  const [value = [], setStorage] = useLocalStorage<string[]>(LOCAL_STORAGE_KEY, []);
+
+  const pushRecentlyUsedDataSource = (ds: DataSourceInstanceSettings) => {
+    if (ds.meta.builtIn) {
+      // Prevent storing the built in datasources (-- Grafana --, -- Mixed --,  -- Dashboard --)
+      return;
+    }
+    if (value.includes(ds.uid)) {
+      // Prevent storing multiple copies of the same data source, put it at the front of the array instead.
+      value.splice(
+        value.findIndex((dsUid) => ds.uid === dsUid),
+        1
+      );
+      setStorage([ds.uid, ...value]);
+    } else {
+      setStorage([ds.uid, ...value].slice(0, 5));
+    }
+  };
+
+  return { recentlyUsedDataSources: value, pushRecentlyUsedDataSource };
```

Comment:
Should we return them as array, as other hooks do? This allows renaming the props depending on the context.